frontend/lib/Views/AdminPage/users/all_users_view.dart

```diff
@@ -0,0 +1,270 @@
+import 'package:adpv_frontend/Views/AdminPage/utils.dart';
+import 'package:adpv_frontend/Widgets/common_widgets.dart';
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+
+import '../../../Repository/AdminRepository/users_list_repository.dart';
+import '../confirmation_dialog_modal.dart';
+
+class ArgsContainer {
+  UserListData userListData;
+  AllUsersListProvider provider;
+
+  ArgsContainer(this.userListData, this.provider);
+}
+
+class AllUsersView extends StatefulWidget {
+  final UsersListRepository repository = UsersListRepository();
+
+  AllUsersView({Key? key}) : super(key: key);
+
+  @override
+  State<AllUsersView> createState() => _AllUsersViewState();
+}
+
+class _AllUsersViewState extends State<AllUsersView> {
+  final _selections = [true, false];
+  late Future<List<UserListData>> users;
+
+  @override
+  void initState() {
+    super.initState();
+    users = widget.repository.getAllUsers();
+  }
+
+  @override
+  Widget build(BuildContext context) =>
+      Scaffold(appBar: buildAdminAppBar("Users list"), body: _buildBody());
+
+  Widget _buildBody() => FutureBuilder<List<UserListData>>(
+        future: users,
+        builder: (context, snapshot) {
+          if (snapshot.connectionState != ConnectionState.done) {
+            return loadingInCenter();
+          } else {
+            return ChangeNotifierProvider(
+              create: (context) => AllUsersListProvider(users),
+              child: Consumer<AllUsersListProvider>(
+                builder: (context, provider, _) => RefreshIndicator(
+                  onRefresh: () => _onPullDownRefresh(provider),
+                  child: Column(
+                    children: [
+                      _buildSortBar(provider),
+                      Expanded(
+                        child: _buildList(provider),
+                      ),
+                    ],
+                  ),
+                ),
+              ),
+            );
+          }
+        },
+      );
+
+  ListView _buildList(AllUsersListProvider provider) => ListView(
+        padding: const EdgeInsets.symmetric(
+          horizontal: 20,
+        ),
+        children:
+            provider.usersList.map((e) => _buildUserCard(e, provider)).toList(),
+      );
+
+  void _onDeletePressed(ArgsContainer args) {
+    showAlertDialog(
+      context,
+      'Delete ' + args.userListData.email + '?',
+      "You are about to delete this user",
+      () => {
+        widget.repository.deleteUser(args.userListData.id).then((value) {
+          users = widget.repository.getAllUsers();
+          users.then((value) => args.provider.init(value));
+        })
+      },
+    );
+  }
+
+  void _editUser(ArgsContainer args) {
+    editUserRoleDialog(context, args.userListData, args.provider);
+  }
+
+  Future<void> _onPullDownRefresh(AllUsersListProvider provider) async {
+    users = widget.repository.getAllUsers();
+    await users.then((value) => provider.init(value));
+  }
```

Comment:
Jeśli to możliwe, to prosiłbym o uporządkowanie kolejności funkcji pod względem ich wywoływania w tym pliku. Jak już napisałem wyżej, łatwiej się czyta kod, jeżeli możemy to robić do góry do dołu bez zbędnego skakania :smile_cat:

Reply:
akurat tutaj jest uporządkowane, mam wrażenie że tylko do mnie się czepiasz o to. Na samym dole klasy wywalone części wspólne które w przyszłości stąd wylecą. Jedyne co nie jest w KSIĄŻKOWEJ kolejności, to funkcja _buildUserCard. Cała reszta jest dobrze. Komentarz do statystyk

Reply:
Rzuciło mi się w oczy, że po `_buildList` powinna być funkcja `_buildUserCard`, natomiast jest `_onDeletePressed`. Głównie dlatego się ten komentarz pojawił